python/common/org/python/types/Float.java

```diff
@@ -737,6 +777,84 @@ public void __delitem__(org.python.Object other) {
         throw new org.python.exceptions.TypeError("'float' object does not support item deletion");
     }
 
+    @org.python.Method(
+        __doc__ = ""
+    )
+    public org.python.types.Tuple as_integer_ratio() {
+        // Get the Mantissa and Exponent
+        MantissaExponent me = frexp(this.value);
+        for (int i=0; i < 300 && me.mantissa != Math.floor(me.mantissa); i++) {
+            me.mantissa *= 2.0;
+            me.exponent--;
+        }
+        // This is dubious, we should replicate the behaviour of PyLong_fromDouble which appears
+        // to perform non-trivial reasoning about rounding.
+        // TODO: implement org.python.types.int.fromDouble, replicate CPython rounding behaviour
+        if (Double.isInfinite(me.mantissa)) {
+            throw new org.python.exceptions.OverflowError("cannot convert float infinity to integer");
+        } else if (Double.isNaN(me.mantissa)) {
+            throw new org.python.exceptions.ValueError("cannot convert NaN to integer");
+        }
+        // Get the Numerator and Denominator
+        org.python.types.Int numerator = new org.python.types.Int((long) me.mantissa);
+        org.python.types.Int denominator = new org.python.types.Int((long) 1);
+        org.python.types.Int pyexponent = me.exponent < 0
+            ? new org.python.types.Int(-me.exponent)
+            : new org.python.types.Int(me.exponent);
+        if (me.exponent > 0) {
+            numerator = (org.python.types.Int) numerator.__lshift__(pyexponent);
+        } else {
+            denominator = (org.python.types.Int) denominator.__lshift__(pyexponent);
+        }
+        // Getting the answer as an arraylist to return
+        java.util.ArrayList<org.python.Object> list = new java.util.ArrayList<org.python.Object>();
+        list.add(numerator);
+        list.add(denominator);
+        return new org.python.types.Tuple(list);
+    }
+
+    /**
+     * Private Helper Function
+     * Breaks down a floating-point value into it's component mantissa and exponent.
+     *
+     * @param x - double, the value of the number that needs to be broken down.
+     * @return MantissaExponent - pair containing the mantissa and exponent.
+     *
+     * @notes
+     *  Code Stolen from Jython
```

Comment:
"stolen" isn't a nice word. Jython's license allows re-use, so what we're doing here isn't criminal. "Based on a version from Jython" is a more appropriate phrasing.

Reply:
Sure, :stuck_out_tongue:!